Add excerpts to drawers

jxjj opened the pull request:
This allows users to add excerpts of video and audio files to drawers, and view the excerpts.

## Details
When a user clickes the "Add to Drawer" button, if the file is excerptable the modal will show a checkbox:
<img width="400" alt="ScreenShot 2023-07-28 at 19 53 02@2x" src="https://github.com/UMN-LATIS/elevator-ui/assets/980170/8f8fbf99-a6cf-42d9-b865-6e6d2454e025">

When checked, the user can scrub to a location and click SET to set the start or end time, or they can enter the times in timestring format (seconds, MM:SS, HH:MM:SS, etc):
<img width="400" alt="ScreenShot 2023-07-28 at 19 53 49@2x" src="https://github.com/UMN-LATIS/elevator-ui/assets/980170/19305f89-208e-48b5-93f4-c8c25c46ad38">

The iframe within the modal is an`<ExcerptableIframe>` component, which handles sending/receiving messages parent window and `jwplayer` within iframe (e.g. updating the parent with the curentScrubberPosition)..

The Excerpt section does some basic input validation, if a user selects a drawer AND fills in text for creating a drawer:
<img width="400" alt="ScreenShot 2023-07-28 at 20 30 05@2x" src="https://github.com/UMN-LATIS/elevator-ui/assets/980170/8d0ad4ec-e1f0-4bb2-ba82-70114cd523df">

Or if they're missing some required data for the excerpt:
<img src="https://github.com/UMN-LATIS/elevator-ui/assets/980170/1da894cd-3f17-4627-a2d4-df7d31d4b72c" width="400">

The "Add to Drawer" button is disabled until all fields are valid. When adding an excerpt, the excerpt must be at least 1 second long to be valid.

If, for some reason, the start time is after the end time, the submit function will swap the values so that they're in the correct order (I triggered a server error or playback issues if these values weren't valid).

When added to the drawer, an excerpt will show the correct excerpt label (not the asset title), and an "Excerpt" chip to differentiate it from other assets within the drawer.
<img src="https://github.com/UMN-LATIS/elevator-ui/assets/980170/b823a2e5-bbf8-4791-b99e-aebba35d2020" width="500" />

The excerpt page layout is simple, showing the excerpt label, start/end times, a "View Asset" link, and buttons for downloading, sharing, and viewing details:
<img width="600" alt="ScreenShot 2023-07-28 at 20 17 58@2x" src="https://github.com/UMN-LATIS/elevator-ui/assets/980170/0e83838e-80ba-4fa5-acba-66cd53c7d2dd">

Again, the excerpt page will load an `<ExcerptableIframe>`, which will send a message to the child iframe  to `setPlaybackBounds`.  I'm currently logging child/parent messages to the console. I'd like to leave this in for now to help with debugging.

The RemoveFromDrawer button was updated to handle removing excerpts, which is a slightly different api call. Confirm modals and toast were also updated to show the excerpt label and not the asset title:

<img width="400" alt="ScreenShot 2023-07-28 at 20 24 02@2x" src="https://github.com/UMN-LATIS/elevator-ui/assets/980170/3e5810d5-4bed-4c07-9d64-f02ee67e4203">

Finally, I added some subtle iconography to video and audio thumbnails, which made it a bit easier for me to tell which files should be excerptable:
<img src="https://github.com/UMN-LATIS/elevator-ui/assets/980170/803fde6c-6c25-408e-bce9-256d385bda38" width="300" />

Closes:
- #171 
- #133 
- and maybe #205 (there's no longer a Create Drawer button at all)

cmcfadden commented on src/components/ThumbnailImage/ThumbnailImage.vue:
```diff
@@ -15,27 +15,47 @@
   }
 );
+
+const isVideo = computed((): boolean => {
```
If we’re going to use file extensions here, they should mirror the allow list types from audiohandler/moviehandler.

cmcfadden reviewed (commented):
One thing I found in testing - if you start the movie playing (in the main window) then open the drawer modal, you're getting events from both players so the times get confused. Maybe we can emit a pause when opening the popup

jxjj commented:
Updates:
- file extensions for video and audio match those in elevator
- fixed iframe play/pause issues. I refactored a bit so that I could scope messages to a particular iframe. Now a pause will be sent to the main object viewer when the modal is opened. Similarly, if the modal is cancelled a pause is sent to the the excerpt iframe to prevent it from continuing to run.

On dev for review
